src/Network/GRPC/MQTT/Core.hs

```diff
@@ -28,54 +29,92 @@ import Data.Conduit.Network.TLS (
   runTLSClient,
   tlsClientConfig,
  )
+import Network.Connection (TLSSettings (TLSSettingsSimple))
 import Network.MQTT.Client (
   MQTTClient,
   MQTTConduit,
   MQTTConfig (..),
-  MessageCallback,
+  MessageCallback (NoCallback),
   runMQTTConduit,
  )
+import Network.MQTT.Topic (Filter, Topic (unTopic), mkFilter)
+import Network.MQTT.Types (LastWill, Property, ProtocolLevel (Protocol311))
+import Relude.Unsafe (fromJust)
 import Turtle (NominalDiffTime)
 
 {- |
-  Wrapper around 'MQTTConfig' indicating whether or not to use TLS for the
-  connection to the MQTT broker
+  Superset of 'MQTTConfig'
 -}
-data MQTTConnectionConfig
-  = Secured MQTTConfig
-  | Unsecured MQTTConfig
+data MQTTGRPCConfig = MQTTGRPCConfig
+  { useTLS :: Bool
+  , mqttMsgSizeLimit :: Int
```

Comment:
It would be useful to leave a comment noting the unit of the limit (e.g. I assume this is just bytes).

Reply:
Excellent point. I'll add one.